client/state/plugins/premium/selectors.js

```diff
@@ -34,6 +39,7 @@ export const getPluginsForSite = function( state, siteId, whitelist = false ) {
 	}
 
 	return filter( pluginList, plugin => {
+		// eslint-disable-next-line no-extra-boolean-cast
 		if ( !! whitelist ) {
```

Comment:
Yeah, I know, I know. But please see my other comment above - https://github.com/Automattic/wp-calypso/pull/31684#discussion_r269578673. I'm happy to fix this in a follow-up PR, but for this one I prefer to refrain from doing changes that could break something. The only reason I'm touching this is because it won't let me commit without ignoring the linter (which I hate doing anyway). FWIW, I usually fix those as I go, but in this case I prefer to refrain from it, given the size of this PR.

Reply:
I can't imaging any case where this would change behavior, but if you want to remove the ignore and fix the lint in another PR, that's fine with me.

Reply:
As I mentioned, this can come from a query argument, and as such, it can lead to weird results. I'd very much prefer to improve the handling, e.g. make a whitelist for values of `whitelist` (pun intended), and reconsider any usage of it in the app. Again, work for another PR.

Also, this boils down to how much changes we want to introduce in such a big PR, and I don't want to introduce any unnecessary ones.

Reply:
Some last words on this one :sweat_smile: The `if` branch will be executed if and only if the conditions/value in parens is truthy (not just `true`), thus short-circuiting the `!!`.

I was a bit uncomfortable with seeing this eslint rule in action at first as well, but I think it really only exists because it _is_ a safe thing to remove `!!` in cases like this -- there's a hard-and-fast equivalence of `if ( condiftion )` and `if ( !! condition )` (as opposed to `const value = condition; ` vs `const value = !! condition;`, which are obviously _not_ equivalent).

(The point being that the `if` performs that cast-to-bool implicitly itself, since it only cares about truthiness vs falsiness.)

Does that make sense? :slightly_smiling_face:

Reply:
It all makes sense! I totally hear you, but let me explain again why I didn't do this in this PR:

> Also, this boils down to how much changes we want to introduce in such a big PR, and I don't want to introduce any unnecessary ones.
 
Also, we're all clear that the next obvious step is to remove the `!!`, and the eslint disable only makes it more obvious that it should happen. Again, I'd like that to happen in another PR, where we could make sure we don't break anything in a more targeted effort.